### PR TITLE
[#438] get version of XJC at runtime and dump it on source generation

### LINKED ISSUE #438: Get XJC Version from Mojo
Add code to get XJC /driver/ version when running maven MOJO
Needed for #338

### PR DESCRIPTION
**Pull request by laurentschoelens**
Fixes #438

### CODE REVIEW

File: maven-plugin/plugin-core/src/main/java/org/jvnet/jaxb/maven/AbstractXJCMojo.java [hunk 1]
```diff
@@ -41,6 +41,14 @@
 public abstract class AbstractXJCMojo<O> extends AbstractMojo implements
 		DependencyResourceResolver {
 
+    /**
+     * This method should be overriden in extending classes to get real value
+     * @return currently running XJC version
+     */
+    public XJCVersion getVersion() {
+        return XJCVersion.empty();
```
Comment (mattrpav) [inline maven-plugin/plugin-core/src/main/java/org/jvnet/jaxb/maven/AbstractXJCMojo.java:49]: How about we make this a XJCVersion.NOT_DETECTED; <-- static string?

File: maven-plugin/plugin-core/src/main/java/org/jvnet/jaxb/maven/XJCVersion.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
+package org.jvnet.jaxb.maven;
+
+public class XJCVersion {
+    private String raw = "N/A";
```
Comment (mattrpav) [inline maven-plugin/plugin-core/src/main/java/org/jvnet/jaxb/maven/XJCVersion.java:4]: Perhaps add a public static final String NOT_DETECTED = "NOT DETECTED" <-- as a way to signal to users that there is a tooling version mismatch?

File: maven-plugin/plugin-core/src/main/java/org/jvnet/jaxb/maven/XJCVersion.java [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+            String[] split = version.split("\\.");
+            if (split.length >= 3) {
+                major = Integer.valueOf(split[0]);
+                minor = Integer.valueOf(split[1]);
+                bugfix = Integer.valueOf(split[2]);
+            }
+        }
+    }
+
+    public static XJCVersion empty() {
```
Comment (mattrpav) [inline maven-plugin/plugin-core/src/main/java/org/jvnet/jaxb/maven/XJCVersion.java:25]: don't love this.. seems like a static constant could work instead?

**Review by mattrpav — CHANGES_REQUESTED** [review]

File: maven-plugin/plugin-core/src/main/java/org/jvnet/jaxb/maven/AbstractXJCMojo.java (hunk 1 above)
Comment (laurentschoelens) [reply maven-plugin/plugin-core/src/main/java/org/jvnet/jaxb/maven/AbstractXJCMojo.java:49]: Will define static instance UNDEFINED

File: maven-plugin/plugin-core/src/main/java/org/jvnet/jaxb/maven/XJCVersion.java (hunk 2 above)
Comment (laurentschoelens) [reply maven-plugin/plugin-core/src/main/java/org/jvnet/jaxb/maven/XJCVersion.java:4]: method isKnown will return true if at least one version (major, minor, bugfix) is > 0

**Review by mattrpav — APPROVED** [review]
LGTM

**Pull request merged**